pkgs/by-name/ha/hashrat/package.nix

```diff
@@ -2,43 +2,42 @@
   lib,
   stdenv,
   fetchFromGitHub,
-  fetchpatch,
+  versionCheckHook,
+  nix-update-script,
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "hashrat";
-  version = "1.22";
+  version = "1.23";
 
   src = fetchFromGitHub {
     owner = "ColumPaget";
     repo = "Hashrat";
-    rev = "v${version}";
-    hash = "sha256-mjjK315OUUFVdUY+zcCvm7yeo7XxourR1sghWbeFT7c=";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-dDtn01eoODgRtsbEdtuXhib3fZLosP8ajJBGaG8Skss=";
   };
 
-  patches = [
-    # fix cross compilation by replacing hardcoded ar with AC_PROG_AR
-    # https://github.com/ColumPaget/Hashrat/pull/27
-    (fetchpatch {
-      url = "https://github.com/ColumPaget/Hashrat/commit/a82615e02021245850a1703e613055da2520c8fd.patch";
-      hash = "sha256-tjyhM2ahZBRoRP8WjyQhrI3l20oaqMtfYmOeAZVEZqU=";
-    })
-  ];
-
   configureFlags = [ "--enable-xattr" ];
 
   makeFlags = [ "PREFIX=$(out)" ];
 
-  meta = with lib; {
+  nativeInstallCheckInputs = [ versionCheckHook ];
+  versionCheckProgramArg = "--version";
+  doInstallCheck = true;
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
     description = "Command-line hash-generation utility";
     mainProgram = "hashrat";
     longDescription = ''
       Hashing tool supporting md5,sha1,sha256,sha512,whirlpool,jh and hmac versions of these.
       Includes recursive file hashing and other features.
     '';
     homepage = "http://www.cjpaget.co.uk/Code/Hashrat";
```

Comment:
At least for me this URL does not work. Maybe use the Github Repo as `homepage`

Reply:
Thanks!